lib/_shared/widgets/chat_bubble_widget.dart

```diff
@@ -10,96 +12,110 @@ class ChatBubble extends StatelessWidget {
-                fontWeight: FontWeight.bold),
-          ),
-          // original language
-          Text(
-            message,
-            style: Theme.of(context)
-                .textTheme
-                .bodyMedium
-                ?.copyWith(color: Colors.white),
+          color: isMine ? Colors.black26 : Colors.black87,
+        ),
+        padding: const EdgeInsets.all(8.0),
+        child: BlocBuilder<ChatLanguageBloc, ChatLanguageState>(
+          builder: (context, state) => Column(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.start,
+            crossAxisAlignment:
+                isMine ? CrossAxisAlignment.end : CrossAxisAlignment.start,
+            children: [
+              // display name
+              Text(
+                displayName ?? 'Unknown',
+                style: Theme.of(context).textTheme.bodyMedium?.copyWith(
+                    color: isMine ? Colors.black87 : Colors.grey,
+                    fontWeight: FontWeight.bold),
+              ),
+              // original language
+              Text(
+                message,
+                style: Theme.of(context)
+                    .textTheme
+                    .bodyMedium
+                    ?.copyWith(color: Colors.white),
+              ),
+              // english version (if there is)
+              if (translations.isNotEmpty && state is ChatLanguageChanged)
+                ...translations.entries
+                    .where(
+                      (element) => element.key == state.selectedLanguage.code,
+                    )
+                    .map(
+                      (e) => Text.rich(
+                        TextSpan(children: [
+                          TextSpan(
+                              text: '${e.key} ',
+                              style: Theme.of(context)
+                                  .textTheme
+                                  .bodySmall
+                                  ?.copyWith(
+                                      fontWeight: FontWeight.bold,
+                                      color: isMine
+                                          ? Colors.black87
+                                          : Colors.grey)),
+                          TextSpan(
+                            text: e.value,
+                            style: Theme.of(context)
+                                .textTheme
+                                .bodySmall
+                                ?.copyWith(
+                                    fontStyle: FontStyle.italic,
+                                    color:
+                                        isMine ? Colors.black87 : Colors.grey),
+                          )
+                        ]),
+                        textAlign: isMine ? TextAlign.right : TextAlign.left,
+                      ),
+                    )
+            ],
           ),
-          // english version (if there is)
```

Comment:
if this English translation is deleted, where on the UI does the translation show?

Reply:
This is original comment code not relate my edited code, so I will change this comment.